### ISSUE TITLE
Movable Buttons

### ISSUE THREAD

**Issue by Sonwon1** [osmandapp/OsmAnd#18022]
I would like to be able to move all of the buttons. These buttons appear to be unmovable Profile, Search, Compass, Map Linked to Location, Plus and Minus.

I would prefer to have them all across the bottom than have them move around on the screen when I switch profiles from Global to Navigation.

**Comment by franco999** [comment]
I agree with activating the displacement of all buttons at the convenience of each user. For example, since the phones are already so tall, I would like to have the zoom buttons a little more in the middle of the screen for finger comfort when holding it with one hand.

**Comment by Sonwon1** [comment]
The zoom level that flashes on the screen when zooming in or out should also be movable. It is currently flashing under a widget which makes it unreadable and there is no way I see to fix.

**Comment by DmitryAlexei** [comment]
@Sonwon1 could you, please, clarify what is *The zoom level that flashes on the screen when zooming in or out*?

**Comment by Sonwon1** [comment]
> @Sonwon1 could you, please, clarify what is _The zoom level that flashes on the screen when zooming in or out_?

I am not sure what to call it. When I press the plus or minus button on the left hand side of the screen it briefly shows the zoom level above the main menu and navigation buttons placement. Zoomed all of the way out shows 2 + .00 3.3 all the way in shows 22 + .00 3.3. The .00 part of the line changes but then drops down to .00 maybe that has something to do with the number of tiles to draw?

There was a setting somewhere to show it on the screen all of the time, at least that is what I thought, maybe it was a widget. However I do not think so but I could be confused. I also see there is a widget to show the zoom level, that is not enabled until I figure out a way to get rid of the zoom level flashing. Hmm, I wonder if it would go away if I enabled the widget?

**Comment by luxcalor** [comment]
Absolutely yes to this. Also would be nice to be able to switch main UI buttons on and off. For example, I never use zoom buttons - double tap and zoom works flawlessly even in gloves. I'd much prefer to have map configuration/profiles button and main menu button under my right thumb since I use them all the time. I also rarely use navigation button, since I normally navigate along the preplanned route and start from there. And if I ever needed the navigation separately, it is always there in the main menu. So I'd rather move search and "rec" buttons down instead, for easy and fast one-handed access.

**Comment by Sonwon1** [comment]
While we are tweaking the buttons can we make them persistent if we do not want them fading from the screen? I dislike the main menu button and the navigation button fading from the screen while navigating or using the device. I suppose this is mostly useful for tablets.

**Comment by luxcalor** [comment]
> can we make them persistent if we do not want them fading from the screen? 

+ to that

**Comment by luxcalor** [comment]
Is there any chance this could be realized anytime soon? Love the app functionality, but the UI is not exactly suitable for my use cases. @vshcherb

**Comment by vshcherb** [comment]
Hopefully 4.7 release this year but that's doesn't include what it means soon

**Comment by luxcalor** [comment]
> According to our plan in 4.7 it will be possible to change standard buttons, later to have more custom buttons "quick actions".

Hi. I am on 4.7.10 beta I think and could not find an option to customize standard buttons' position and/or appearance. Do I miss something or the possibility is not there yet?

**Comment by vshcherb** [comment]
It was part of big story for buttons customization and only first part of creating custom buttons make through. 2nd and 3rd phase configuring visibility and size wasn't ready Q1 2024, so it's postponed for next release

**Comment by luxcalor** [comment]
> so it's postponed for next release
Any estimations on when it comes out?

**Comment by rm318** [comment]
Hi, is there any progress that the standard Zoom +/-, Compass, My position and Back to menu buttons are also movable?
In the meantime (OsmAnd+ 4.9.7), the widgets are largely customizable and even the size of the zoom buttons is adjustable.

However, I want to use OsmAnd+ on my motorcycle as a replacement for a Garmin navigation device and therefore need the zoom buttons on the left side of the display, as I need the right hand for the throttle and front brake.

**Comment by luxcalor** [comment]
> Hi, is there any progress that the standard Zoom +/-, Compass, My position and Back to menu buttons are also movable?

Looking forward to this myself

**Comment by Sonwon1** [comment]
Me too

**Comment by vshcherb** [comment]
there will be no generic solution, you will need to create custom button with +, -, my location - . As moving standard buttons won't be supported to have long tap present on them

**Comment by Sonwon1** [comment]
> there will be no generic solution, you will need to create custom button with +, -, my location - . As moving standard buttons won't be supported to have long tap present on them

Thank you, I didn't know that was possible now. I am assuming we can also turn off the standard buttons and replace them with custom buttons. I will check it out when I have time. Busy planning a motorcycle trip to Scotland.

**Comment by rm318** [comment]
Hi vshcherb, adding Zoom In/Out as custom buttons is working!

But your explanation with the standard buttons and long tab is not really convincing. Because the standard buttons itself start to moving round if other elements are active ( e.g. an activated left panel will move the Compass element and long tabs are still working.

**Comment by vshcherb** [comment]
Yes that would probably need a specific new request for action My location without zoom. Standard buttons could be hidden though they shouldn't be moved by custom buttons (that's kind of a bug if it happens probably)

> And “Back to menu” can also be hidden, while the function can be activated by swiping from the left edge to the center.
Yes

**Comment by Flavelius** [comment]
Hi, I'm trying to use a custom button to with the search action to work around not being able to move the default search button down (I have a large phone and it's inconvenient having to use my other hand just to tap that button at the top while walking etc.).
But this action only opens a large overlay panel with 3 buttons:
1 does open the actual search
2 allows adding other actions
3 is empty
This unfortunately completely invalidates my use case, the main interaction still happens at the top, and it now even requires an additional tap, plus it feels unnecessary to be able to add new actions in the overly when there is a dedicated editing screen for this.
Can I skip this intermediate popup panel somehow, or can I move the search to the bottom in any other way?

**Comment by vshcherb** [comment]
Yes, if you have a button with single action it shouldn't display other action suggestions. Probably mentionned in documentation

**Comment by Flavelius** [comment]
Thanks for your response. This was my bad, you're right; I was confusing the quick action entry for a default pre-added custom button, until I used the + at the top at which point it became visible that those are separate things.
As this only appeared to me after I questioned my actions following your comment, I want to suggest designing a more prominent visual separation of the quick action from the list of custom button entries, maybe by moving it to the bottom or into a panel-like visual container.
